examples/qml_features/rust/src/externcxxqt.rs

```diff
@@ -0,0 +1,109 @@
+// SPDX-FileCopyrightText: 2022 Klarälvdalens Datakonsult AB, a KDAB Group company <info@kdab.com>
+// SPDX-FileContributor: Andrew Hayzen <andrew.hayzen@kdab.com>
+//
+// SPDX-License-Identifier: MIT OR Apache-2.0
+
+//! This example shows how an external QObject with signals can be used
+
+/// A CXX-Qt bridge which shows how an external QObject with signals can be used
+// ANCHOR: book_cxx_file_stem
+#[cxx_qt::bridge(cxx_file_stem = "externcxxqt")]
+pub mod ffi {
+    unsafe extern "C++Qt" {
+        include!("external_qobject.h");
+        /// ExternalQObject C++ class
+        type ExternalQObject;
+
+        /// Trigger emitting the signal "amount" times
+        fn trigger(self: Pin<&mut ExternalQObject>, amount: u32);
+
+        /// Signal that is emitted when trigger is fired
+        #[qsignal]
+        fn triggered(self: Pin<&mut ExternalQObject>);
+
+        /// Private signal that is emitted when trigger is fired
+        #[qsignal]
+        #[rust_name = "triggered_private_signal"]
+        pub(self) fn triggeredPrivateSignal(self: Pin<&mut ExternalQObject>);
+    }
+
+    unsafe extern "RustQt" {
+        #[qobject]
+        #[qml_element]
+        #[qproperty(u32, count)]
+        #[qproperty(u32, private_count)]
+        type ExternalCxxQtHelper = super::ExternalCxxQtHelperRust;
+
+        #[qinvokable]
+        unsafe fn connect_to_external(
+            self: Pin<&mut ExternalCxxQtHelper>,
+            external: *mut ExternalQObject,
+        );
+
+        #[qinvokable]
+        unsafe fn trigger_on_external(
+            self: Pin<&mut ExternalCxxQtHelper>,
+            external: *mut ExternalQObject,
+            amount: u32,
+        );
+    }
+
+    impl cxx_qt::Threading for ExternalCxxQtHelper {}
+}
+
+use core::pin::Pin;
+
+/// Test struct
+#[derive(Default)]
+pub struct ExternalCxxQtHelperRust {
+    count: u32,
+    private_count: u32,
+}
+
+impl ffi::ExternalCxxQtHelper {
+    unsafe fn connect_to_external(self: Pin<&mut Self>, external: *mut ffi::ExternalQObject) {
+        // let qt_thread = self.qt_thread();
+        if let Some(external) = external.as_mut() {
+            let mut pinned_external = Pin::new_unchecked(external);
+            pinned_external
+                .as_mut()
+                .on_triggered(|_| {
+                    // FIXME: can't move this inside the function pointer
```

Comment:
cc @LeonMatthesKDAB we need the closure support in signals to complete this example :-)